src/bp-core/compatibility/bp-incompatible-plugins-helper.php

```diff
@@ -790,6 +790,25 @@ function bb_get_elementor_maintenance_mode_template() {
 }
 add_action( 'bp_loaded', 'bb_get_elementor_maintenance_mode_template' );
 
+/**
+ * Update the current component and action for elementor saved library preview link.
+ * 
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param object $bp     BuddyPress object.
+ * @param array  $bp_uri Array of URI.
+ * 
+ * @return void
+ */
+function bb_support_elementor_library_preview_permalink( $bp, $bp_uri ) {
+	global $bp_plugins;
+	if ( in_array( 'elementor/elementor.php', $bp_plugins ) && isset( $_GET['elementor_library'] ) ) {
+		$bp->current_component = '';
+		$bp->current_action    = '';
+	}
+}
+add_action( 'bp_core_set_uri_globals', 'bb_support_elementor_library_preview_permalink', 10, 2 );
+
```

Comment:
I would recommend creating a separate file for the elementor related compatibility issue same we did for other plugins.

Reply:
Hi @chetansatasiya , I have created a new file for elementor and moved all elementor compatibility related code there.